classes/class-paysoncheckout-for-woocommerce-ajax.php

```diff
@@ -110,7 +110,16 @@ public static function pco_wc_update_checkout() {
 			);
 		}
 
-		$subscription = ( class_exists( 'WC_Subscriptions_Cart' ) && WC_Subscriptions_Cart::cart_contains_subscription() ) ? true : false;
+		$subscription = false;
+		if ( class_exists( 'WC_Subscriptions_Cart' ) ) {
+			foreach ( WC()->cart->get_cart() as $cart_item ) {
+				$product_id = $cart_item['product_id'];
+				if ( WC_Subscriptions_Product::is_subscription( $product_id ) ) {
+					$subscription = true;
+					break;
+				}
+			}
+		}
```

Comment:
This is not the correct way to check for this. The main issue here is that a purchase can be a subscription without any of the products being a subscription product.

For example there are plugins like [All Products Subscription](https://woocommerce.com/products/all-products-for-woocommerce-subscriptions/) for WooCommerce that makes is so that every product type can have subscriptions enabled. Not all of these plugins modify the products to return as subscription products.

I am however not sure why this would not work for the case when paying for a failed order. But lets discuss this internally a bit more.